src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBinderSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,469 @@
+                int startingIndentationLevel = _writer.IndentationLevel;
+
+                EmitVarDeclarationToDefault(keyType, LiteralFor_key);
+                EmitVarDeclarationToDefault(elementType, LiteralFor_element);
+
+                _writer.WriteBlockStart($"foreach ({_typeSymbolForIConfigurationSection.ToDisplayString()} {LiteralFor_section} in {LiteralFor_configuration}.{LiteralFor_GetChildren}())");
+
+                if (TryEmitBindLogicFromString(keyType, LiteralFor_key, expressionForConfigStringValue: ExpressionFor_sectionKey) &&
+                    TryEmitBindLogicFromIConfigurationSection(elementType, LiteralFor_element, InitializationKind.SimpleAssignment))
+                {
+                    EmitAssignment($@"{LiteralFor_obj}[{LiteralFor_key}]", LiteralFor_element);
+                    _writer.WriteBlockEnd();
+                }
+                else
+                {
+                    int currentLength = _writer.Length;
+                    Debug.Assert(currentLength > startingLength);
+                    _writer.Remove(startingLength, currentLength - startingLength, startingIndentationLevel);
+                    Emit_NotSupportedException_UnableToBindType(spec.Type, NotSupportedReason_CollectionKeyOrElementNotSupported);
+                }
+
+                void EmitVarDeclarationToDefault(ITypeSymbol type, string varName)
+                {
+                    _writer.WriteLine($"{type.ToDisplayString()} {varName} = {DefaultKeyword};");
+                }
+            }
+
+            private static void EmitBindMethodImplForEnumerable(CollectionBindingSpec spec)
+            {
+                Debug.Assert(spec.ElementType is not null);
+            }
+
+            private enum InitializationKind
+            {
+                None = 0,
+                SimpleAssignment,
+                AssignmentWithNullCheck,
+                Declaration,
+            }
+
+            private bool TryEmitObjectInit(ITypeSymbol type, string expressionForMemberAccess, InitializationKind initKind)
+            {
+                if (initKind == InitializationKind.None)
+                {
+                    return true;
+                }
+
+                if (type.IsAbstract || type.TypeKind == TypeKind.Interface)
+                {
+                    Emit_NotSupportedException_UnableToBindType(type, NotSupportedReason_AbstractOrInterfaceNotSupported);
+                }
+                else if (HasPublicParameterlessCtor(type))
+                {
+                    if (initKind == InitializationKind.Declaration)
+                    {
+                        Debug.Assert(!expressionForMemberAccess.Contains("."));
+                        EmitAssignment($"{type.ToDisplayString()} {expressionForMemberAccess}", "new()");
+                    }
+                    else if (initKind == InitializationKind.AssignmentWithNullCheck)
+                    {
+                        if (!type.IsValueType)
```

Comment:
>IsValueType [](http://example.com/codeflow?start=34&length=11)

what happen when we have a value type here? we still return true from TryEmitObjectInit?